rust/kcl-lib/src/unparser.rs

```diff
@@ -794,30 +815,28 @@ impl Node<PipeExpression> {
             .map(|(index, statement)| {
                 let indentation = options.get_indentation(indentation_level + 1);
                 let mut s = statement.recast(options, indentation_level + 1, ExprContext::Pipe);
-                let non_code_meta = self.non_code_meta.clone();
+                let non_code_meta = &self.non_code_meta;
                 if let Some(non_code_meta_value) = non_code_meta.non_code_nodes.get(&index) {
                     for val in non_code_meta_value {
-                        let formatted = if val.end == self.end {
-                            val.recast(options, indentation_level)
-                                .trim_end_matches('\n')
-                                .to_string()
+                        let ind = if val.end == self.end {
+                            indentation_level
                         } else {
-                            val.recast(options, indentation_level + 1)
-                                .trim_end_matches('\n')
-                                .to_string()
+                            indentation_level + 1
                         };
+                        let formatted = val.recast(options, ind);
+                        let formatted = formatted.trim_end_matches('\n');
                         if let NonCodeValue::BlockComment { .. } = val.value {
                             s += "\n";
-                            s += &formatted;
+                            s.to_mut().push_str(formatted);
                         } else {
-                            s += &formatted;
+                            s.to_mut().push_str(formatted);
                         }
                     }
                 }
```

Comment:
Rather than have multiple `to_mut()` calls, we could do `let s = s.to_mut()` shadowing at the start of this block?